client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -327,158 +309,198 @@ export async function addGeneToSetAndExpand(
 }
 
 /**
  * GENE
  */
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function addGeneToSet(genesetName: any, geneToAddToSet: any) {
+export async function addGeneToSet(
+  genesetName: string,
+  geneToAddToSet: string,
+  page: Page
+): Promise<void> {
   const submitButton = `${genesetName}:submit-gene`;
-  await clickOnUntil(`${genesetName}:add-new-gene-to-geneset`, async () => {
-    await expect(page).toMatchElement(getTestId("add-genes"));
-  });
-  assert(await isElementPresent(getTestId("add-genes"), {}));
-  await typeInto("add-genes", geneToAddToSet);
-  await clickOn(submitButton);
+  await page.getByTestId(`${genesetName}:add-new-gene-to-geneset`).click();
+  await tryUntil(
+    async () => {
+      await page
+        .getByTestId("add-genes-to-geneset-dialog")
+        .fill(geneToAddToSet);
+      expect(page.getByTestId(submitButton)).toBeEnabled();
+    },
+    { page }
+  );
+  await page.getByTestId(submitButton).click();
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function removeGene(geneSymbol: any) {
+export async function removeGene(
+  geneSymbol: string,
+  page: Page
+): Promise<void> {
   const targetId = `delete-from-geneset:${geneSymbol}`;
 
-  await clickOn(targetId);
+  await page.getByTestId(targetId).click();
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function assertGeneExistsInGeneset(geneSymbol: any) {
-  const handle = await waitByID(`${geneSymbol}:gene-label`);
-
-  // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
-  const result = await handle.evaluate((node) =>
-    node.getAttribute("aria-label")
-  );
+export async function assertGeneExistsInGeneset(
+  geneSymbol: string,
+  page: Page
+): Promise<void> {
+  const result = await page
+    .getByTestId(`${geneSymbol}:gene-label`)
+    .getAttribute("aria-label");
 
   return expect(result).toBe(geneSymbol);
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-export async function assertGeneDoesNotExist(geneSymbol: any) {
-  // @ts-expect-error ts-migrate(2554) FIXME: Expected 2 arguments, but got 1.
-  const result = await isElementPresent(getTestId(`${geneSymbol}:gene-label`));
+export async function assertGeneDoesNotExist(
+  geneSymbol: string,
+  page: Page
+): Promise<void> {
+  await tryUntil(
+    async () => {
+      const geneLabel = await page.getByTestId(`${geneSymbol}:gene-label`);
+      await geneLabel.waitFor({ state: "hidden" });
+      await expect(geneLabel).toBeHidden();
```

Comment:
do we need this if the previous step is already checking it?

Reply:
technically no, but it felt wrong to have a function `assertGeneDoesNotExist` without an actual assertion

Reply:
Got it! Sounds good 👍